src/Symfony/Component/Security/Core/Validator/Constraints/UserPassword.php

```diff
@@ -29,8 +32,32 @@ class UserPassword extends Constraint
     public $message = 'This value should be the user\'s current password.';
     public $service = 'security.validator.user_password';
 
-    public function __construct(array $options = null, string $message = null, string $service = null, array $groups = null, mixed $payload = null)
-    {
+    public function __construct(
+        /*
+         * Options to pass the validation constraint.
+         */
+        array $options = null,
+
+        /*
+         * The message to display if this constraint raises a violation.
+         */
+        string $message = null,
+
+        /*
+         * The service id to use to check the user current password.
+         */
+        string $service = null,
+
+        /*
+         * The groups on which this validation constraint applies.
+         */
+        array $groups = null,
+
+        /*
+         * Arbitrary data to be attached to the constraint.
```

Comment:
I don't fully understand what `attached` means here. Maybe we can use a simpler verb like: "included", "passed", etc. Also, maybe we can explain very concisely the purpose of this?

E.g.

```
Arbitrary data passed to the constraint. This is useful for XXX or YYYY.
```